### PR TITLE
Fix message counter

### PR DESCRIPTION
**Pull request by timofey-barmin**
If writecomb is on message counter doesn't work right in some cases because we write more than one message at a time. This PR fixes my case but I'm not sure about the following case in tcpkali_engine.c:

```
2132         if(token) {
2133             if(position <= token && token < position + wrote) {
2134                 conn->traffic_ongoing.msgs_sent++;
2135             }
2136         }
```

### CODE REVIEW

**Comment by timofey-barmin** [comment]
... and btw it fixes https://github.com/machinezone/tcpkali/issues/49

File: src/tcpkali_engine.c [hunk 1]
```diff
@@ -2134,7 +2134,7 @@ latency_record_outgoing_ts(TK_P_ struct connection *conn, const void *new_positi
                 conn->traffic_ongoing.msgs_sent++;
             }
         } else {
-                conn->traffic_ongoing.msgs_sent++;
+            conn->traffic_ongoing.msgs_sent += wrote / conn->data.single_message_size;
```
Comment (vlm) [inline src/tcpkali_engine.c]: This is incorrect fix. If a message is sent piece by piece (long message with a very low --rate), the msgs_sent won't increase at all.

Comment (timofey-barmin) [reply src/tcpkali_engine.c]: Yeah, that's exactly why it was closed

File: src/tcpkali_expr.c [hunk 2]
```diff
@@ -310,3 +316,24 @@ unescape_expression(tk_expr_t *expr) {
...
+    case EXPR_REGEX:
+        return tregex_avg_size(expr->u.regex.re);
+    case EXPR_DATA:
+    case EXPR_RAW:
+    case EXPR_MODULO:
+    case EXPR_CONNECTION_PTR:
+    case EXPR_CONNECTION_UID:
+    case EXPR_MESSAGE_MARKER:
+    case EXPR_WS_FRAME: {
+        return expr->estimate_size;
```
Comment (vlm) [inline src/tcpkali_expr.c]: This will overshoot for CONNECTION_UID.

File: src/tcpkali_expr_y.y [hunk 3]
```diff
@@ -202,7 +202,7 @@ NumericExpr:
     | TOK_message '.' TOK_marker {
         $$ = calloc(1, sizeof(*($$)));
         $$->type = EXPR_MESSAGE_MARKER;
-        $$->estimate_size = sizeof("100000000000000" "1000000000000000" "!");
+        $$->estimate_size = sizeof("1000000000000" "1000000000000000" "!") - 1;
```
Comment (vlm) [inline src/tcpkali_expr_y.y]: Why?

Comment (timofey-barmin) [reply src/tcpkali_expr_y.y]: because actually the length of the marker is 30 and not 33 bytes

File: src/tcpkali_expr.c (hunk 2 above)
Comment (timofey-barmin) [reply src/tcpkali_expr.c]: Yes, agree. Need to fix it

File: src/tcpkali_regex.c [hunk 4]
```diff
@@ -66,6 +67,7 @@ tregex_debug_print(tregex *re) {
         printf(")");
         break;
     case TRegexRepeat:
+        tregex_debug_print(re->repeat.what);
```
Comment (vlm) [inline src/tcpkali_regex.c]: Do you need it here?

**Comment by vlm** [comment]
Looks OK. However, operation test (.sh) fail. Would you please fix it?

**Pull request merged**